{"name": "test_login_pos_01", "status": "failed", "statusDetails": {"message": "AssertionError: Expected URL to contain 'salah', but got 'https://www.saucedemo.com/inventory.html'", "trace": "browser_chrome = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"6aaf1716d0953486dc6d83037a2ffd06\")>\n\n    def test_login_pos_01(browser_chrome):\n        \"\"\"\n        TC_ID: LOGIN_POS_01\n        Title: Login with valid user\n        Expected: Redirected to inventory page\n        \"\"\"\n        login = LoginPage(browser_chrome)\n        login.open()\n        login.login(Config.STANDARD_USER, Config.PASSWORD)\n        take_screenshot(browser_chrome, step_name=\"input_pos01\")\n        # validasi dengan assert import dari module assertion + take screeshot dari utils.helpers\n        try:\n            url_contains(browser_chrome, \"salah\")\n            take_screenshot(browser_chrome, step_name=\"result_pos01_as expected\")\n        except AssertionError as e:\n            take_screenshot(browser_chrome, step_name=\"result_pos01_unexpected\")\n>           raise e\n\ntests\\test_login.py:25: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\ntests\\test_login.py:21: in test_login_pos_01\n    url_contains(browser_chrome, \"salah\")\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"6aaf1716d0953486dc6d83037a2ffd06\")>\nkeyword = 'salah'\n\n    def assert_url_contains(driver, keyword):\n        # verifikasi URL saat ini mengandung kata tertentu\n        actual_result = driver.current_url\n        expected_result = keyword\n        # tambahan if untuk menampilkan pesan jika true\n        if expected_result in actual_result:\n            print(f\"\\n {expected_result} was found in {actual_result}\")\n        # assert memvalidasi kebenaran dan menghentikan eksekusi jika salah\n>       assert expected_result in actual_result, f\"Expected URL to contain '{expected_result}', but got '{actual_result}'\"\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\nE       AssertionError: Expected URL to contain 'salah', but got 'https://www.saucedemo.com/inventory.html'\n\nutils\\assertions.py:9: AssertionError"}, "description": "\nTC_ID: LOGIN_POS_01\nTitle: Login with valid user\nExpected: Redirected to inventory page\n", "attachments": [{"name": "screenshot_input_pos01", "source": "9388fccd-2095-4a1c-bacb-b5a57c398ca0-attachment.png", "type": "image/png"}, {"name": "screenshot_result_pos01_unexpected", "source": "2cf2dd83-5bed-4597-bf5e-0c06e0b64373-attachment.png", "type": "image/png"}], "start": 1759695335656, "stop": 1759695339581, "uuid": "c8257dd5-4e9c-443e-9957-1b38e2041d3f", "historyId": "f7dd2152c6cfa5665aa1b92919645c42", "testCaseId": "f7dd2152c6cfa5665aa1b92919645c42", "fullName": "tests.test_login#test_login_pos_01", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_login"}, {"name": "host", "value": "DESKTOP-N1PMKPL"}, {"name": "thread", "value": "26896-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_login"}], "titlePath": ["tests", "test_login.py"]}